src/Concerns/CapturesState.php

```diff
@@ -72,7 +72,8 @@ public function configureSampling(string $by): void
         Compatibility::addHiddenContext('nightwatch_should_sample', $this->shouldSample);
 
         if (! $this->shouldSample) {
-            $this->state->records->flush();
+            $this->state->flush();
+            $this->ingest->flush();
```

Comment:
If we always flush both the state and ingest, I wonder if it's worth adding the `flush` method to the `Core` class that does this in one go.

Reply:
I can get around that idea! I'm gonna merge this one while it is approved. Might add this in a follow up PR, though!